### PR TITLE
Add chaos collision meshes for UE5

### PR DESCRIPTION
**Pull request by nithinp7**
Fixes compilation errors when targeting UE5 and adds collision mesh generation when the Chaos physics engine is being used.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @nithinp7!

* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium-unreal/blob/ue5-support-nithin/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] [Cesium for Unreal Samples](https://github.com/CesiumGS/cesium-unreal-samples) works.

File: Source/CesiumRuntime/Private/CesiumGltfComponent.cpp [hunk 1]
```diff
@@ -1971,23 +1955,136 @@ void UCesiumGltfComponent::FinishDestroy() {
...
+        true,
+        pCollisionMesh);
+  }
+}
+
+#else
+template <typename TIndex>
+static bool isDegenerate(
+    const Chaos::TVector<TIndex, 3>& triangle,
+    const TArray<FStaticMeshBuildVertex>& vertexData) {
```
Comment (nithinp7) [inline Source/CesiumRuntime/Private/CesiumGltfComponent.cpp:2002]: I don't think this check is actually required, this came from trying to build meshes similar to how Unreal does it, but in our case we can be somewhat confident going in that our triangles are not ill-formed after the tiling process. Would you agree @kring?

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @nithinp7! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @nithinp7! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by kring** [comment]
This all looks good to me @nithinp7. Just need to fix the UE 4.26 build errors. I'm going to do a build of the latest UE5 to see how it works there, but it doesn't need to hold up this PR.

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @nithinp7! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Pull request merged**